ruby/ql/lib/codeql/ruby/security/UnsafeDeserializationCustomizations.qll

```diff
@@ -48,12 +48,15 @@ module UnsafeDeserialization {
   }
 
   /**
-   * An argument in a call to `YAML.load`, considered a sink for unsafe
-   * deserialization.
+   * An argument in a call to `YAML.load` or `YAML.load_file`, considered a sink
+   * for unsafe deserialization. As the `YAML` module is an alias of `Psych` in
    */
   class YamlLoadArgument extends Sink {
     YamlLoadArgument() {
-      this = API::getTopLevelMember("YAML").getAMethodCall("load").getArgument(0)
+      this =
+        API::getTopLevelMember(["YAML", "Psych"])
+            .getAMethodCall(["load", "load_file"])
```

Comment:
I think I'm ok with treating `load_file` as a sink, but it strikes me as far less serious than `load`, since it's only an unsafe-deserialization problem if there's some existing dangerous YAML file on the server. I vaguely recall that that's why I didn't originally include it.

Reply:
Yeah, I agree that this case seems far less severe in most cases. I had a quick look at the JS and Python versions for precedent. In those cases the `load` like methods seem to be able to take plain data or some file stream. However, I'm not sure if the argument would be consider to be tainted if the file input stream path was user-controlled?

Reply:
As discussed in the weekly meeting, I've removed the `load_file` arg as a sink and just kept the `Psych` change.